app/models/concerns/orchestration/dhcp.rb

```diff
@@ -19,6 +20,20 @@ def dhcp_record
     @dhcp_record ||= (provision? && jumpstart?) ? Net::DHCP::SparcRecord.new(dhcp_attrs) : Net::DHCP::Record.new(dhcp_attrs)
   end
 
+  def rebuild_dhcp
+    begin
+      if dhcp?
+        set_dhcp
+      else
+        logger.info "DHCP not supported for #{host.name}, #{ip}, skipping orchestration rebuild"
+        true
+      end
+    rescue => e
```

Comment:
Nitpick, can you move this begin/rescue block around the set_* method specifically, ditto in all other orchestration types.  It avoids the danger of us catching an exception in surrounding code and misreporting it.